src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) object;
+            return (color != null && color.equals(rectangle.color))
+                    || (color == null && rectangle.color == null)
+                    && (width != null && width == rectangle.width)
+                    || (width == null && rectangle.width == null)
+                    && (length != null && length == rectangle.length)
+                    || (length == null && rectangle.length == null);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 14;
+        result = 28 * result + (color == null ? 0 : color.hashCode());
+        result = 28 * result + (width == null ? 0 : width);
+        result = 28 * result + (length == null ? 0 : length);
```

Comment:
why 14 and 28?

Reply:
I dont know, its random numbers. Bohdan in the video choose 17 and 31. I was shure that i can choose whatever i want either.

Reply:
i have added 31 and 17

Reply:
You can choose any number but it should be a primary one and 31 is the most popular